In [1]:
import pandas as pd
url = "https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2020/2020-05-05/villagers.csv"
df = pd.read_csv(url)
df.isna().sum()

row_n           0
id              1
name            0
gender          0
species         0
birthday        0
personality     0
song           11
phrase          0
full_id         0
url             0
dtype: int64

In [2]:
# Assuming you have already loaded the dataset into a DataFrame called villagers_df

# Get the number of rows and columns
num_rows, num_columns = villagers_df.shape

# Print the result
print(f'The dataset has {num_rows} rows and {num_columns} columns.')


NameError: name 'villagers_df' is not defined

In [3]:
# Assuming you have already loaded the dataset into a DataFrame called villagers_df

# Get the number of rows and columns
num_rows, num_columns = df.shape

# Print the result
print(f'The dataset has {num_rows} rows and {num_columns} columns.')


The dataset has 391 rows and 11 columns.


In [4]:
# General summary statistics for numerical columns
df.describe()


row_n
count  391.000000
mean   239.902813
std    140.702672
min      2.000000
25%    117.500000
50%    240.000000
75%    363.500000
max    483.000000

In [5]:
# Summary statistics for categorical columns
df.describe(include='object')


id     name gender species birthday personality          song  \
count       390      391    391     391      391         391           380   
unique      390      391      2      35      361           8            92   
top     admiral  Admiral   male     cat     1-27        lazy  K.K. Country   
freq          1        1    204      23        2          60            10   

         phrase           full_id  \
count       391               391   
unique      388               391   
top     wee one  villager-admiral   
freq          2                 1   

                                                      url  
count                                                 391  
unique                                                391  
top     https://villagerdb.com/images/villagers/thumb/...  
freq                                                    1

In [6]:
# Count the number of occurrences for each species
df['species'].value_counts()


species
cat          23
rabbit       20
frog         18
squirrel     18
duck         17
dog          16
cub          16
pig          15
bear         15
mouse        15
horse        15
bird         13
penguin      13
sheep        13
elephant     11
wolf         11
ostrich      10
deer         10
eagle         9
gorilla       9
chicken       9
koala         9
goat          8
hamster       8
kangaroo      8
monkey        8
anteater      7
hippo         7
tiger         7
alligator     7
lion          7
bull          6
rhino         6
cow           4
octopus       3
Name: count, dtype: int64

In [7]:
df['species'].value_counts(normalize=True)

species
cat          0.058824
rabbit       0.051151
frog         0.046036
squirrel     0.046036
duck         0.043478
dog          0.040921
cub          0.040921
pig          0.038363
bear         0.038363
mouse        0.038363
horse        0.038363
bird         0.033248
penguin      0.033248
sheep        0.033248
elephant     0.028133
wolf         0.028133
ostrich      0.025575
deer         0.025575
eagle        0.023018
gorilla      0.023018
chicken      0.023018
koala        0.023018
goat         0.020460
hamster      0.020460
kangaroo     0.020460
monkey       0.020460
anteater     0.017903
hippo        0.017903
tiger        0.017903
alligator    0.017903
lion         0.017903
bull         0.015345
rhino        0.015345
cow          0.010230
octopus      0.007673
Name: proportion, dtype: float64

In [8]:
url = "https://raw.githubusercontent.com/mwaskom/seaborn-data/master/titanic.csv"
df_titanic = pd.read_csv(url)

In [9]:
df_titanic.shape

(891, 15)

In [10]:
df_titanic.describe()

survived      pclass         age       sibsp       parch        fare
count  891.000000  891.000000  714.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.699118    0.523008    0.381594   32.204208
std      0.486592    0.836071   14.526497    1.102743    0.806057   49.693429
min      0.000000    1.000000    0.420000    0.000000    0.000000    0.000000
25%      0.000000    2.000000   20.125000    0.000000    0.000000    7.910400
50%      0.000000    3.000000   28.000000    0.000000    0.000000   14.454200
75%      1.000000    3.000000   38.000000    1.000000    0.000000   31.000000
max      1.000000    3.000000   80.000000    8.000000    6.000000  512.329200

In [11]:
df_titanic.columns

Index(['survived', 'pclass', 'sex', 'age', 'sibsp', 'parch', 'fare',
       'embarked', 'class', 'who', 'adult_male', 'deck', 'embark_town',
       'alive', 'alone'],
      dtype='object')

### Summary of Interaction:

During our interaction, the following key tasks and concepts were discussed and addressed:

1. **Dataset Overview**: 
   - The dataset about Animal Crossing characters (villagers) was loaded, consisting of **391 rows** (observations) and **11 columns** (variables).
   - We clarified that **observations** refer to individual entries (i.e., each villager), and **variables** refer to the attributes of those villagers (such as species, personality, etc.).

2. **Dataset Analysis in Python**:
   - You were provided with Python code to get the shape of the dataset in JupyterHub using `villagers_df.shape`, which confirmed the dataset's dimensions.
   - The usage of the `df.describe()` method was explained to summarize numeric and categorical columns. This method provides a quick statistical overview of the dataset, such as count, mean, standard deviation, min, max, and quartile values for numeric data, or the count and frequency of categories for non-numeric data.
   - The `df['column'].value_counts()` method was explained for counting occurrences of unique values in specific columns (e.g., counting how many villagers belong to each species). Additionally, the option to normalize counts to percentages using `normalize=True` was discussed.

3. **Key Code Examples**:
   - Summarizing numeric columns: `villagers_df.describe()`
   - Summarizing categorical columns: `villagers_df.describe(include='object')`
   - Counting unique values in a column: `villagers_df['species'].value_counts()`
   - Getting percentages for unique values: `villagers_df['species'].value_counts(normalize=True)`

This interaction provided practical guidance for exploring and summarizing datasets in Python, particularly focusing on using pandas functions for analyzing both numeric and categorical data.